test/distributed/test_symmetric_memory.py

```diff
@@ -41,6 +42,9 @@
 )
 
 
+test_contexts = [nullcontext, _test_mode]
```

Comment:
Curious, what would these contexts imply?

Reply:
Good question. The first one is going to be used to dispatch to the real implementation (symmetric memory) and the second one is used to dispatched to a fallback implementation, which doesn't use symmetric memory. I'll add some comments in the next PR. I'll land this PR to unblock @danielvegamyhre to test dim=1 case.